projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/cast/LLVMToI8Node.java

```diff
@@ -132,6 +132,11 @@ public byte executeI8(LLVM80BitFloat from) {
     public abstract static class LLVMAddressToI8Node extends LLVMToI8Node {
```

Comment:
It is because each node that produces a value is an `LLVMExpressionNode` which has an execute method that returns an `Object`. This method exists so that function calls can easily execute the children to get a generic value for each of them.

I still think that it is confusing that an address to I8 (and other integer types) cast node can receive a boolean input value (which we could make transparent by having the `LLVMTo*CastNode`) and that, if we want to integrate the change, should have a test case for it.

If this issue blocks you we can merge the change and handle it properly the next time we stumble on such a case.